src/WebJobs.Script.WebHost/Models/AzureStorageInfoValue.cs

```diff
@@ -83,7 +83,12 @@ public static AzureStorageInfoValue FromEnvironmentVariable(KeyValuePair<string,
             }
 
             var parts = environmentVariable.Value?.Split(new[] { '|' }, StringSplitOptions.RemoveEmptyEntries);
-            if (parts == null || parts.Length != 4)
+            if (parts == null)
+            {
+                return null;
+            }
+
+            if (parts.Length != 4 && parts.Length != 5)
             {
```

Comment:
let's split condition for 4 and 5 and add following logs
1. for four parts - Did not find expected number of parameters to mount file share. Expected - accountname, sharename, accesskey and mountPath. 
2. for 5 parts -  Did not find expected number of parameters to mount file share. Expected - accountname, sharename, accesskey ,mountPath and protocol

Reply:
This file doesn't have `ILogger` access :(

Reply:
:( lets track injecting ILogger here in a seperate PR.